quiz.js

```diff
@@ -1,168 +1,137 @@
+        resetQuestionZone();        
+        
         let currentQuestion = questions[questionIndex];
-        //set correctAnswer equal to the correct answer of the current question
-        correctAnswer = currentQuestion.correct_answer;
-        //tells user what question number they are on
+        correctAnswer       = fix(currentQuestion.correct_answer);
+
         questionNumberEl.text(`Question: ${questionIndex + 1} / 10`)
-        //tells user score
-        scoreEl.text(`Score: ${score}`);
-        //shows user the current question after a fix
-        let fixedCurrentQuestion = fix(currentQuestion.question)
-        questionEl.text(`Question: ${fixedCurrentQuestion}`);
-        //shows user what category the current question is from
-        categoryEl.text(`Category: ${currentQuestion.category}`);
-        //sets up array of all possible answer choices using the spread operator for incorrect answers
-        console.log(currentQuestion.incorrect_answers)
-        for (var i = 0; i < 2; i++) {
-            currentQuestion.incorrect_answers[i] = fix(currentQuestion.incorrect_answers[i]);
-          }
-        // let fixedIncorrectAnswers = fix(currentQuestion.incorrect_answers)
-        let fixedCorrectAnswer = fix(currentQuestion.correct_answer)
-        let questionChoices = [...currentQuestion.incorrect_answers, fixedCorrectAnswer]
-        //shuffle array randomly
-        let randomQuestionChoices = questionChoices.sort(() => Math.random() - 0.5);
-        //creates a temporary btn for each of the choices and adds them to the answer zone; using jQuery adds text, value, class, & onclick event to each btn
-        randomQuestionChoices.forEach(function (value) {
-            let tempBtn = $('<button>').text(value).val(value).addClass('hollow button answer').click(validateAnswer);
-            answerZoneEl.append(tempBtn);
+        scoreEl         .text(`Score: ${score}`);
+        questionEl      .text(`Question: ${fix(currentQuestion.question)}`);
+        categoryEl      .text(`Category: ${currentQuestion.category}`);
+
+        let allBtns = createAnswerButtons(currentQuestion)
+            .sort(() => Math.random() -.5);
+            
+        allBtns.forEach(function(btn) {
+            answerZoneEl.append(btn);
+        })
+    }
+
+    function createAnswerButtons(question) {
+        let buttons = []
+        question.incorrect_answers.forEach(function (value) {
+            let tempBtn = $('<button>')
+                .text(fix(value))
+                .val(fix(value))
+                .addClass('hollow button answer')
+                .click(() => validateAnswer(false));
+                buttons.push(tempBtn);
         })
 
+        let correctBtn = $('<button>')
+            .text(question.correct_answer)
+            .val(question.correct_answer)
+            .addClass('hollow button answer')
+            .click(() => validateAnswer(true));
+        buttons.push(correctBtn);
+        return buttons;
```

Comment:
This may need some explaining:

- For each incorrect answer, create a button for it (but don't append it to the answer zone).
- For the correct answer, create a button for it (but don't append it to the answer zone)
- Put all of those buttons in one list

`validateAnswer(true/false)` is a bit confusing so I'll explain that. 

When we're creating these buttons, we *know* which one is correct and which ones are incorrect. Since we know this, we can tell the `click()` handler what to do without having to do any string checks to see if the button is in fact the correct answer. `validateAnswer()` is wrapped around either `correctAnswerClicked()` or `incorrectAnswerClicked()` with some other logic for incrementing the question counter and waiting to display next question. So when the correct button is clicked, we will call `validateAnswer()` with true as the parameter which will increment score, play cheer etc, and alternatively, calling it with false is done for incorrect answers which'll jeer etc.

The caller of this method then randomly shuffles the buttons and adds them to the answerZone.

Reply:
Oh one more thing about the lambda `() =>` these are equivalent
```
click(function() {
    validateAnswer(true);
})

click(() => validateAnswer(true));
```